tests/parsing/CliParsingTest.py

```diff
@@ -71,3 +71,8 @@ def test_check_conflicts(self):
         with self.assertRaisesRegex(SystemExit, '2') as cm:
             check_conflicts(sections)
             self.assertEqual(cm.exception.code, 2)
+
+        sections = parse_cli(arg_list=['--no-config', '-config'])
```

Comment:
I think that here as well it should be `--config` as, if I'm not mistaken, even though the test is passing it isn't actually checking that functionality. The command fails with an exception, but it fails because it doesn't recognize `-config`. I think it should be `--config` and a .coafile should be provided as argument to the `--config` option. I don't know which is the best way to do this: maybe create a temporary .coafile? As in: https://github.com/coala/coala/blob/ac5aa59e7fa27ce13fca57ed40218477e568307a/tests/parsing/ConfParserTest.py#L45-L47

Reply:
Yes, it should be `--config` but I don't think a temporary .coafile is required. The test is meant to only check whether conflicting arguments are given.

Reply:
In this way I think it would fail because it doesn't have a configuration file provided before checking the conflict between `--config` and `--no-config`. So, if I am right the test will pass but will still not do the intended checking. On the other hand I don't know if a .coafile is needed, any file should do in the end indeed, @pareksha. What are your thoughts on this?

Reply:
looking at the other tests I don't think it is required.